configuration/template/stack.go

```diff
@@ -316,6 +319,32 @@ func MakeUtilFuncMap(varStack map[string]string) map[string]interface{} {
 			}
 			return *in
 		},
+		"HostnameWithin": func(hostname string, prefix string, idMinStr string, idMaxStr string) string {
```

Comment:
Just a suggestion: perhaps we could call it `SuffixWithin` or `SuffixInRange` or somesuch? I realize you use it for hostnames but there's no reason to assume the input string must be a hostname.

Reply:
Agreed. Initially I went for `HostnameWithin` so it is easier for users to realize what it could be used for, but if we have anyway an example of TPC, it should be clear enough.